pkg/webhook/mutation/pod/otlp/mutator.go

```diff
@@ -0,0 +1,234 @@
+
+	envVarDtApiToken = "DT_API_TOKEN"
+
+	otlpAPIPath = "/v2/otlp"
+)
+
+type OTLPMutator struct{}
+
+func NewMutator() dtwebhook.Mutator {
+	return &OTLPMutator{}
+}
+
+func (OTLPMutator) IsEnabled(request *dtwebhook.BaseRequest) bool {
+	//TODO implement when structure for OTLP enrichment config in Dynakube has been defined.
+	return true
+}
+
+func (OTLPMutator) IsInjected(request *dtwebhook.BaseRequest) bool {
+	// TODO the check for already present env vars is also done in the Mutate() implementation, so I'm not sure we need this check here as well.
+	// Maybe a check for an annotation indicating that the OTLP vars have already been injected might be better here.
+	otlpEndpointEnvVars := []string{envVarOtlpExporterEndpoint, envVarOtlpTraceEndpoint, envVarOtlpMetricsEndpoint, envVarOtlpLogsEndpoint}
+	// if any of the OTEL_EXPORTER_* variables are already set in all containers, return false, as we do not want to overwrite them
+	// TODO check if also initContainers should be considered
+	gotOtlpExporterEnvVars := 0
+
+	for _, c := range request.Pod.Spec.Containers {
+		found := false
+		for _, e := range c.Env {
+			// check if any of the otlp exporter env vars is already present in the container
+			if slices.Contains(otlpEndpointEnvVars, e.Name) {
+				found = true
+				break
+			}
+		}
+		if found {
+			gotOtlpExporterEnvVars++
+		}
+	}
+	if gotOtlpExporterEnvVars == len(request.Pod.Spec.Containers) {
+		return true
+	}
+	return false
+}
+
+func (OTLPMutator) Mutate(request *dtwebhook.MutationRequest) error {
+	injectCert := false
+	if request.DynaKube.ActiveGate().HasCaCert() {
+		injectCert = true
+		// mount the volume with the AG certificate
+		defaultMode := int32(420)
+		agCertVolume := corev1.Volume{
+			Name: activeGateTrustedCertVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					DefaultMode: &defaultMode,
+					SecretName:  consts.OtlpIngestCertsSecretName,
+				},
+			},
+		}
+		request.Pod.Spec.Volumes = append(request.Pod.Spec.Volumes, agCertVolume)
+	}
+
+	apiURL, err := otlp.GetOtlpIngestEndpoint(&request.DynaKube)
+	if err != nil {
+		return fmt.Errorf("could not acquire ingest endpoint", err)
+	}
+
+	// TODO check if also initContainers should be considered
+	for i := range request.Pod.Spec.Containers {
+		envVarsToCheck := []string{envVarOtlpExporterEndpoint, envVarOtlpExporterHeaders, envVarOtlpExporterCertificate, envVarOtlpExporterProtocol}
+		for _, envVar := range envVarsToCheck {
+			// TODO how to handle EnvFrom? OTLP env vars could also be imported via a configMap
+			if isEnvVarSet(request.Pod.Spec.Containers[i].Env, envVar) {
+				continue
+			}
+		}
+		// TODO use secret specifically for otlp
+		// TODO the creation of this secret via the SecretGenerator used by the Dynakube reconciler can take quite some time (~5mins when tested locally), due to the reconciliation interval
```

Comment:
This is the reason the ["replicate" logic exists](https://github.com/Dynatrace/dynatrace-operator/blob/cee6877a2e4de9e12c8634ca9447fc7c9e6dd62d/pkg/webhook/mutation/pod/handle.go#L156)

The idea is that the `SecretGenerator` maintains 2 "types of" Secrets:
- A "source" Secret, within the dynatrace namespace
- Secrets, in the namespaces that exist at time of reconcile

This "replicate" is for namespaces that are created within this time window between reconciles.
- So the webhook doesn't have to regen the secret on the fly, just copy it over from the "source" Secret